.github/workflows/ci.yml

```diff
@@ -6,77 +6,76 @@ on:
+
+      - name: cache mvn
+        uses: actions/cache@v4
         with:
           path: ~/.m2/repository
-          key: maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: maven-
-      - name: store branch name in environment
-        run: echo "BRANCH_NAME=$(echo ${GITHUB_REF##*/})" >> "$GITHUB_ENV"
-      - name: git config
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: release preparation
+        id: prep
         run: |
           git config --global user.name "github-actions[bot]"
           git config --global user.email "github-actions[bot]@users.noreply.github.com"
-      - name: configure maven
-        run: |
-          mkdir -p "$HOME/.m2"
-          cat <<EOF > "$HOME/.m2/settings.xml"
-          <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
-            <servers>
-              <server>
-                <id>github</id>
-                <username>\${env.GITHUB_ACTOR}</username>
-                <password>\${env.GITHUB_TOKEN}</password>
-              </server>
-            </servers>
-          </settings>
-          EOF
+          major_version="$(sed -n "s/^.*<major-version>\([0-9]\{1,\}\)<\/major-version>.*$/\1/p" pom.xml)"
+          echo "$major_version" | grep -q "^[0-9]\{1,\}$" || {
+            echo "ERROR: Failed to extract <major-version> from pom.xml"
+            exit 1
+          }
+          echo "major-version=$major_version" >> "$GITHUB_OUTPUT"
+
+      - uses: capralifecycle/actions-lib/generate-tag@v1
+        id: tag
+        with:
+          tag-prefix: "${{ steps.prep.outputs.major-version }}"
+          tag-type: "punctuated-timestamp-tag"
+
+      - name: resolve dependencies
+        env:
+          # NOTE: Use the following token if your project does not depend on private packages:
+          # GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # NOTE: Use the following token if your project depends on private packages:
+          GITHUB_TOKEN: ${{ secrets.SHARED_GITHUB_PACKAGES_TOKEN }}
+        run: mvn -B dependency:resolve
+
       - name: conditional release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          export MAVEN_OPTS="-Duser.home=$HOME" # Home directory not automatically picked up for some reason
-          major_version="$(sed -n "s/^.*<major-version>\([0-9]\{1,\}\)<\/major-version>.*$/\1/p" pom.xml)"
-          date="$(date -u "+%Y%m%d.%H%M%S")"
-          tag="$major_version.$date"
-          printf "Tag to use for potential release: %s\n" "$tag"
-          mvn -B org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce -Drules=requireReleaseDeps
+          SONAR_TOKEN: ${{ secrets.SHARED_SONAR_TOKEN }}
```

Comment:
SonarCloud isn't currently configured for this repository, so I suggest we remove this for now

Reply:
https://github.com/capralifecycle/liflig-properties/pull/57/commits/8931fdaa803e68be86832c15d6528b7a87510012